athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchHelper.java

```diff
@@ -0,0 +1,734 @@
+
+    // Predicate conjunctions.
+    private static final String AND_OPER = " AND ";
+    private static final String OR_OPER = " OR ";
+    private static final String EMPTY_PREDICATE = "";
+
+    // Existence predicates.
+    private static final String existsPredicate(boolean exists, String fieldName)
+    {
+        if (exists) {
+            // (_exists:field)
+            return "(_exists_:" + fieldName + ")";
+        }
+        else {
+            // (NOT _exists_:field)
+            return "(NOT _exists_:" + fieldName + ")";
+        }
+    }
+
+    private ElasticsearchHelper() {}
+
+    /**
+     * Get an environment variable using System.getenv().
+     * @param var is the environment variable.
+     * @return the contents of the environment variable or an empty String if it's not defined.
+     */
+    public static final String getEnv(String var)
+    {
+        String result = System.getenv(var);
+
+        return result == null ? "" : result;
+    }
+
+    /**
+     * @return the boolean value of autoDiscoverEndpoint.
+     */
+    public static final boolean isAutoDiscoverEndpoint()
+    {
+        return autoDiscoverEndpoint;
+    }
+
+    /**
+     * Gets a client factory that can create an Elasticsearch REST client injected with credentials.
+     * @return a new client factory.
+     */
+    public static AwsRestHighLevelClientFactory getClientFactory()
+    {
+        logger.info("getClientFactory - enter");
+
+        autoDiscoverEndpoint = getEnv(AUTO_DISCOVER_ENDPOINT).equalsIgnoreCase("true");
+        secretName = getEnv(SECRET_NAME);
+
+        if (autoDiscoverEndpoint) {
+            // Client factory for clients injected with AWS credentials.
+            return AwsRestHighLevelClientFactory.defaultFactory();
+        }
+        else {
+            // Client factory for clients injected with username/password credentials.
+            return new AwsRestHighLevelClientFactory.Builder().withSecretCredentials(secretName).build();
+        }
+    }
+
+    /**
+     * Sets the domainMap with domain-names and corresponding endpoints retrieved from the AWS ES SDK.
+     * @param domainStatusList is a list of status objects returned by a listDomainNames request to the AWS ES SDK.
+     */
+    private static final void setDomainMapping(List<ElasticsearchDomainStatus> domainStatusList)
+    {
+        logger.info("setDomainMapping(List<>) - enter");
+
+        domainMap.clear();
+        for (ElasticsearchDomainStatus domainStatus : domainStatusList) {
+            domainMap.put(domainStatus.getDomainName(), domainStatus.getEndpoint());
+        }
+
+        logger.info("setDomainMapping(List<>) - exit: " + domainMap);
+    }
+
+    /**
+     * Sets the domainMap with domain-names and corresponding endpoints stored in a Map object.
+     * @param mapping is a map of domain-names and their corresponding endpoints.
+     */
+    public static final void setDomainMapping(Map<String, String> mapping)
```

Comment:
Why is this static vs being an instance on the appropriate handler?

Reply:
parsing common and state in handler. Duplication is okay.

Reply:
Made ElasticsearchHelper a singleton rather than a static class. Removed static where applicable.

Reply:
Gone due to refactoring.